tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,748 @@
+"""Test the Broadlink config flow."""
+import errno
+import socket
+
+import broadlink.exceptions as blke
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.broadlink.const import DOMAIN
+
+from . import get_device
+
+from tests.async_mock import call, patch
+
+
+@pytest.fixture(autouse=True)
+def broadlink_setup_fixture():
+    """Mock broadlink entry setup."""
+    with patch(
+        "homeassistant.components.broadlink.async_setup_entry", return_value=True
+    ):
+        yield
+
+
+async def test_flow_user_works(hass):
+    """Test a config flow initiated by the user.
+
+    Best case scenario with no errors or locks.
+    """
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)